src/App.tsx

```diff
@@ -8,37 +8,39 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { setAuthor } from './features/author';
 import { User } from './types/User';
+import { clearPosts } from './features/posts';
+import { setSelectedPost as setSelectedPostAction } from './features/selectedPost';
 import { Post } from './types/Post';
+import { changeAsync } from './features/users';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
-
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const dispatch = useAppDispatch();
+  const { author } = useAppSelector(state => state.author);
+  const { posts, loading, error } = useAppSelector(state => state.posts);
+  const { selectedPost } = useAppSelector(state => state.selectedPost);
 
   function loadUserPosts(userId: number) {
-    setLoaded(false);
-
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
+    dispatch(changeAsync(userId));
   }
 
+  const setCurrentAuthor = (newAuthor: User) => {
+    dispatch(setAuthor(newAuthor));
+  };
+
+  const handleSetSelectedPost = (post: Post | null) => {
+    dispatch(setSelectedPostAction(post));
+  };
+
   useEffect(() => {
-    // we clear the post when an author is changed
-    // not to confuse the user
-    setSelectedPost(null);
+    dispatch(setSelectedPostAction(null));
 
     if (author) {
       loadUserPosts(author.id);
     } else {
-      setPosts([]);
+      dispatch(clearPosts());
     }
   }, [author]);
```

Comment:
just add dispatch here

![image](https://github.com/mate-academy/react_redux-list-of-posts/assets/86923089/cfa39d1b-2858-48f5-9645-6a23773127a8)